src/makim/makim.py

```diff
@@ -93,6 +95,16 @@ def _call_shell_app(self, cmd):
             _new_session=True,
         )
 
+        self.working_directory_scoped = None  # Reset scoped working directory
```

Comment:
instead of setting working directory here,
please do it inside the change_target and change_group_data methods.
use the shell_app flow as a reference: 
- https://github.com/osl-incubator/makim/blob/main/src/makim/makim.py#L138
- https://github.com/osl-incubator/makim/blob/main/src/makim/makim.py#L162-L163
- https://github.com/osl-incubator/makim/blob/main/src/makim/makim.py#L169C1-L170